groups.go

```diff
@@ -0,0 +1,121 @@
+package hue
+
+import (
+	"encoding/json"
+	"net/http"
+)
+
+type Group struct {
+	bridge *Bridge
+
+	ID     string
+	Name   string
+	Lights []string
+	Type   string
+	Class  string
+	Action LightState
+}
+
+// Groups returns the service to interact with the groups on this bridge.
+func (b *Bridge) Groups() *GroupsService { return &GroupsService{bridge: b} }
+
+// GroupsService is the service that allows interacting with the groups API
+// of the bridge.
+type GroupsService struct{ bridge *Bridge }
+
+func (g *GroupsService) idMap() (groups map[string]*Group, err error) {
+	msg, err := g.bridge.call(http.MethodGet, nil, "groups")
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
Let's remove all the empty lines in this method to keep it consistent with other files...